jax/_src/xla_bridge.py

```diff
@@ -787,12 +787,15 @@ def _discover_and_register_pjrt_plugins():
   _alias_to_platforms.setdefault(_alias, []).append(_platform)
 
 
+def known_platforms() -> set[str]:
+  return set(_backend_factories.keys()) | set(_platform_aliases.values())
```

Comment:
There's a PR adding a lowering for `METAL` right now. But we don't know about Apple metal in-tree, there's a plugin.

I suggest just having a set of known backends included here also?

Reply:
I believe `register_plugin` populates `_backend_factories`, doesn't that cover this case?

> I suggest just having a set of known backends included here also?

Do you mean adding an informal list of backend "friends" here?

Reply:
@hawkinsp should `METAL` be added here?
https://github.com/google/jax/blob/51e743139b2bbe5146e257610eca31a05f277f83/jax/_src/xla_bridge.py#L227
Then `_nonexperimental_plugins` could be added to the set.

Reply:
No, because the role of `_nonexperimental_plugins` is to suppress an "experimental" warning. The METAL plugin is currently experimental.

Reply:
I took the liberty of adding a set of `_experimental_plugins` with the goal of tracking experimental out-of-tree plugins that nevertheless have registrations in JAX. What do you think?

Reply:
@hawkinsp gentle ping

Reply:
I think you need to add "rocm" to this list also. I'm also not sure about the case of these, so maybe canonicalize the case or do a case insensitive match?

Reply:
Actually never mind I guess you get that from the backend factories list.